app/recipe/serializers.py

```diff
@@ -0,0 +1,44 @@
+from rest_framework import serializers
+
+from core.models import Ingredient, Recipe
+
+
+class IngredientSerialiser(serializers.ModelSerializer):
+    """Serializer for ingredient objects"""
+
+    class Meta:
+        model = Ingredient
+        fields = ['name']
+
+
+class RecipeDetailSerializer(serializers.ModelSerializer):
+    """Serialize a recipe"""
+    ingredients = IngredientSerialiser(many=True)
+
+    class Meta:
+        model = Recipe
+        fields = ['id', 'name', 'description', 'ingredients']
+        read_only_fields = ('id',)
+
+    def create(self, validated_data):
+        ingredients = validated_data.pop('ingredients')
+        recipe_created = Recipe.objects.create(**validated_data)
+        for ingredient in ingredients:
+            Ingredient.objects.create(
+                name=ingredient['name'],
+                recipe=recipe_created
+            )
+        return recipe_created
+
+    def update(self, instance, validated_data):
+        instance.ingredients.all().delete()
+        new_ingredients = validated_data.pop('ingredients')
+        instance.name = validated_data.get('name', instance.name)
+        instance.description = validated_data.get(
+            'description',
+            instance.description
+        )
```

Comment:
this can be deleted, it will be done by the parent update method

Reply:
oh sorry, I didn't get that before. I'll update now :)